corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js

```diff
@@ -56,6 +58,7 @@ hqDefine("data_dictionary/js/data_dictionary", [
         const recurseChunks = function (nextUrl) {
             $.getJSON(nextUrl, function (data) {
                 setCaseProperties(data.groups);
+                self.resetSaveButton();
```

Comment:
Is there a specific reason for resetting the save button after loading each chunk of case properties? If a user makes any changes while all the case properties are being loaded then these changes will be lost. 

I imagine this won't be a problem most of the time but if a domain has tons of properties that takes a while to load then the user might start making changes while the properties are still being loaded.

Reply:
Loading the case properties is treated as a change.
* If we don't reset the save button at all, then it will be enabled, and the user will be prompted to save before leaving the page.
* If we wait until all the case properties have been loaded, e.g.
  ```python
                setCaseProperties(data.groups);
                nextUrl = data._links.next;
                if (nextUrl) {
                    recurseChunks(nextUrl);
                } else {
                    self.resetSaveButton();
                }
  ```
  then the save button is enabled for a second or two while the properties are loaded, and then reset. This looks odd, and potentially allows a user to click the button before the properties have finished loading.
* If we reset the save button with every chunk, the button appears to remain constantly "saved".